src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+import {
+  SelectableBox,
+  Icon,
+  Spinner,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { ArrowDropDown, ArrowDropUp } from '@edx/paragon/icons';
+import PropTypes from 'prop-types';
+import messages from './messages';
+import './ContentTagsDropDownSelector.scss';
+
+import { useTaxonomyTagsDataResponse, useIsTaxonomyTagsDataLoaded } from './api/hooks/selectors';
+
+const ContentTagsDropDownSelector = ({
+  taxonomyId, level, subTagsUrl,
+}) => {
+  const intl = useIntl();
+  // This array represents the states of the dropdowns on this level
+  // Each index contains whether the dropdown is open/closed
+  // O === Closed
+  // 1 === Open
+  const [dropdownStates, setDropdownStates] = useState([]);
+
+  const isOpen = (i) => (dropdownStates[i] === 1);
+
+  const clickAndEnterHandler = (index) => {
+    const updatedState = dropdownStates.map((dropdownState, i) => {
+      if (index === i) {
+        return dropdownState === 0 ? 1 : 0;
+      }
+      return dropdownState;
+    });
+
+    setDropdownStates(updatedState);
+  };
+
+  const useTaxonomyTagsData = (id, fullPathProvided) => {
+    const taxonomyTagsData = useTaxonomyTagsDataResponse(id, fullPathProvided);
+    const isTaxonomyTagsLoaded = useIsTaxonomyTagsDataLoaded(id, fullPathProvided);
+    return { taxonomyTagsData, isTaxonomyTagsLoaded };
+  };
+
+  const { taxonomyTagsData, isTaxonomyTagsLoaded } = useTaxonomyTagsData(taxonomyId, subTagsUrl);
+
+  useEffect(() => {
+    if (isTaxonomyTagsLoaded && taxonomyTagsData) {
+      // When the data first loads, all the dropdowns are initially closed
+      setDropdownStates(new Array(taxonomyTagsData.results.length).fill(0));
+    }
+  }, [isTaxonomyTagsLoaded, taxonomyTagsData]);
+
+  return (
+    isTaxonomyTagsLoaded && taxonomyTagsData
+      ? taxonomyTagsData.results.map((taxonomyTag, i) => (
+        <div style={{ display: 'flex', flexDirection: 'column', paddingLeft: `${level * 1}rem` }}>
```

Comment:
You should use classes for some of this. 
d-flex flex-column 

As for the padding, if it is nested and each keeps a margin you won't need to increase padding at each level?

Reply:
I've replaced them with classes as you suggested. However I kept the `paddingLeft` since it doesn't add any margin/padding (even though it is nested) so they all appear on the same level without it.